java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -102,19 +102,17 @@ class AndroidSqliteDatabaseManager(
             .forEach { it.reset() }
     }
 
-    override suspend fun removeExpiredEntities(): Job = coroutineScope {
-        launch {
-            registry.fetchAll()
-                .map { it.name to getDatabase(it.name, it.isPersistent) }
-                .forEach { (name, db) ->
-                    if (databaseSizeTooLarge(name)) {
-                        // If the database size is too large, we clear it entirely.
-                        db.removeAllEntities()
-                    } else {
-                        db.removeExpiredEntities()
-                    }
+    override suspend fun removeExpiredEntities() {
```

Comment:
why is this better? (add to the pr description)
do you want to do the same for runGarbageCollection()?